packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -552,7 +552,7 @@ class Navigator extends StatefulWidget {
   ///
   /// Returns a [Future] that completes to the `result` value passed to [pop]
   /// when the pushed route is popped off the navigator.
-  static Future<dynamic> push(BuildContext context, Route<dynamic> route) {
+  static Future<T> push<T>(BuildContext context, Route<T> route) {
```

Comment:
this should be marked `@optionalTypes` or whatever it is so that when the T is `dynamic` (which is the common case), it doesn't have to be specified.

Reply:
@pq Should this work for generic methods?  I just tried @optionalTypeArgs and it didn't seem to do the trick.  Is that the right annotation?  Or does it need to go somewhere other than the line before the function declaration?

Reply:
@bwilkerson?

Reply:
The CL that added the annotation to 'meta' doesn't contain any implementation in 'analyzer', nor can I find any reference in 'analyzer' to `optionalTypeArgs`. As far as I can tell, the semantics of this annotation were never implemented.

> Should this work for generic methods?

I can't speak to the _intended_ semantics of the annotation, but the documentation for the annotation reads:
```
Used to annotate a class declaration `C`.
```
If we want to extend that to support generic methods, typedefs, etc. then the documentation should be updated.

Reply:
@optionalTypeArgs definitely works for some things. We use it, and removing it adds analyzer errors.

Reply:
I think this is a linter thing:

https://github.com/dart-lang/linter/blob/cd13719c0b2e71c66f12073fecf03951d0ccafcb/lib/src/rules/always_specify_types.dart

Reply:
@Hixie I'm not sure how to resolve this, shall I just put @optionalTypeArgs on it, and when/if the linter gets fixed to handle this the call sites can be updated?

Reply:
How hard would it be to fix the linter?

Reply:
@bwilkerson : if you look at `always_specify_types` in the linter you can see some special casing that takes `optionalTypeArgs` into consideration.  I'm not sure how hard this would be to fix in the linter but my guess is not very.  @leafpetersen : would you mind opening a bug (and then adding a reference to it in the source) so we know to do the cleanup down the road?

Reply:
@pq A linter bug or a flutter bug?

Reply:
@leafpetersen:  [linter](https://github.com/dart-lang/linter/issues/new) 👍